components/script/script_thread.rs

```diff
@@ -3871,6 +3882,17 @@ impl ScriptThread {
         }
     }
 
+    fn handle_media_session_action(
+        &self,
+        browsing_context_id: TopLevelBrowsingContextId,
+        action: MediaSessionActionType,
+    ) {
+        match self.media_sessions.borrow().get(&browsing_context_id) {
```

Comment:
Have you considered sending the `PipelineId` along in the message, and then grabbing the corresponding document via [`find_document`](https://github.com/servo/servo/blob/882dcc8408159bed5633337d11e83b6839f97e34/components/script/script_thread.rs#L459), and then only handling the action if the document is fully active?

Reply:
On second thoughts, this might be best dealt with via the concept of the [active-media-session](https://w3c.github.io/mediasession/#active-media-session), although I assume the active session will belong to a fully-active document...